powerauth-java-crypto/src/test/java/com/wultra/security/powerauth/crypto/lib/v4/sharedsecret/SharedSecretHybridTest.java

```diff
@@ -109,4 +109,31 @@ public void testEcdheMlkemWithTestVectors(Map<String, String> vector) throws Exc
         assertEquals(Base64.getEncoder().encodeToString(sharedSecret.getEncoded()), vector.get("sharedSecret"));
     }
 
+    @Test
+    public void generateTestVectors() throws Exception {
+        final List<Map<String, String>> vectors = new java.util.ArrayList<>();
+        for (int i = 0; i < 100; i++) {
+            SharedSecretHybrid sharedSecretHybrid = new SharedSecretHybrid();
+            RequestCryptogram request = sharedSecretHybrid.generateRequestCryptogram();
+            SharedSecretRequestHybrid clientRequest = (SharedSecretRequestHybrid) request.getSharedSecretRequest();
+            SharedSecretClientContextHybrid clientContext = (SharedSecretClientContextHybrid) request.getSharedSecretClientContext();
+            ResponseCryptogram serverResponse = sharedSecretHybrid.generateResponseCryptogram(clientRequest);
+            SecretKey derivedSharedSecret = sharedSecretHybrid.computeSharedSecret(
+                    clientContext,
+                    (SharedSecretResponseHybrid) serverResponse.getSharedSecretResponse()
+            );
+            Map<String, String> vector = Map.of(
+                    "ecClientPrivateKey", Base64.getEncoder().encodeToString(KEY_CONVERTOR_EC.convertPrivateKeyToBytes(clientContext.getEcPrivateKey())),
+                    "pqcClientPrivateKey", Base64.getEncoder().encodeToString(KEY_CONVERTOR_PQC.convertPrivateKeyToBytes(clientContext.getPqcKemDecapsulationKey())),
+                    "ecServerPublicKey", ((SharedSecretResponseHybrid) serverResponse.getSharedSecretResponse()).getEcServerPublicKey(),
+                    "pqcEncapsulation", ((SharedSecretResponseHybrid) serverResponse.getSharedSecretResponse()).getPqcEncapsulation(),
```

Comment:
The `pqcEncapsulation` is a bit misleading. It's `ciphertext` in NIST terminology or "wrapped key" (OpenSSL)